src/content/en/updates/2018/07/page-lifecycle-api.md

```diff
@@ -0,0 +1,962 @@
+      <p>When the page changes from <em>active</em> to <em>passive</em>, it's a
+      good time to persist unsaved application state.</p>
+    </td>
+  </tr>
+  <tr>
+    <td><strong>Hidden</strong></td>
+    <td>
+      <p>When the page changes from <em>passive</em> to <em>hidden</em>, it's
+      possible the user will not interact with it again until it's reloaded.</p>
+      <p>The transition to <em>hidden</em> is also often the last state change
+      that's reliably observable by developers (this is especially true on
+      mobile, as users can close tabs or the browser app itself, and the
+      pagehide and unload events are not fired in those cases).</p>
+      <p>This means you should treat the hidden state as the likely end to the
+      user's session &mdash; including persisting any unsaved application state,
+      and sending any unsent analytics data.</p>
+      <p>You should also stop making any UI updates (since they won't be seen
+      by the user), and you should stop any tasks that a user wouldn't want
+      running in the background.</p>
+    </td>
+  </tr>
+  <tr>
+    <td><strong>Frozen</strong></td>
+    <td>
+      <p>In the frozen state, tasks in the task queue are suspended until the
+      page is unfrozen &mdash; which may never happen (e.g. if the page is
+      discarded).</td>
+      <p>This means when the page changes from <em>hidden</em> to <em>frozen</em>
+      it's essential that you stop any timers or tear down any connections that,
+      if frozen, could affect other open tabs in the same origin, or affect the
+      browser's ability to put the page in the
+      <a href="https://webkit.org/blog/427/webkit-page-cache-i-the-basics/">
+      page navigation cache</a>.<p>
+      <p>It's particularly important that you:</p>
+      <ul>
+        <li>Close all open
+          <a href="https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API">
+          IndexedDB</a> connections.</li>
+        <li>Close open
+          <a href="https://developer.mozilla.org/en-US/docs/Web/API/Broadcast_Channel_API">
+          BroadcastChannel</a> connections.</li>
+        <li>Close active <a href="https://developer.mozilla.org/en-US/docs/Web/API/WebRTC_API">
+          WebRTC</a> connections.</li>
+        <li>Stop any network polling or close any open
+          <a href="https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API">
+          Web Socket</a> connections.</li>
+        <li>Release any held <a href="https://github.com/inexorabletash/web-locks">
+          Web Locks</a>.</li>
+      </ul>
+      <p>You should also persist any dynamic view state (e.g. scroll position
+      in an infinite list view) to sessionStorage (or
+      <a href="#save-data-to-indexeddb-before-freezing">IndexedDB via
+      <code>commit()</code></a>) that you'd want restored if the page was
+      discarded and then reloaded later.</p>
+      <p>If the page then transitions from <em>frozen</em> back to
```

Comment:
Delete 'then'.
Use &lt;code> instead of &lt;em>. Please fix throughout.

Reply:
The states aren't really code, nor technically part of the API, so I don't think `<code>` makes sense for them.

I can remove the `<em>` if you think they're not appropriate. I know some people mark up technical terms using `<i>`.